### PR TITLE
Various clean ups to get dummy backends working properly

### PR DESCRIPTION
**Pull request by bloodearnest**
- update config for nhsd and test backends to add new USING_DUMMY_DATA_BACKEND
 - added forgotten nhsd test
 - pull all docker images on first install
 - update test Dockerfile to use new core-packages.txt

### CODE REVIEW

File: tests/nhsd.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -euo pipefail
+./nhsd-backend/manage.sh
+#run again to check for idempotency
+./nhsd-backend/manage.sh
```
Comment (evansd) [inline tests/nhsd.sh:5]: I love (in an entirely non-ironic way) the fact that the test is just to do the same thing twice and check it doesn't blow up.

File: scripts/jobrunner.sh [hunk 2]
```diff
@@ -71,6 +70,14 @@ done
 chown -R jobrunner:jobrunner "$HIGH_PRIVACY_STORAGE_BASE"
 chown -R jobrunner:jobrunner "$MEDIUM_PRIVACY_STORAGE_BASE"
 
+# ensure docker images present
+# Note: does not update, as that is currently done manually
+
+for image in cohortextractor python jupyter r base-docker busybox; do
```
Comment (evansd) [inline scripts/jobrunner.sh:76]: Not a point for now, but I wonder if we should think about moving the list of allowed images out of the job-runner Python code and into some easily readable config file so we can reuse it here.

**Review by evansd — APPROVED** [review]
:+1: Nice

File: tests/nhsd.sh (hunk 1 above)
Comment (bloodearnest) [reply tests/nhsd.sh:5]: I'd like to add some more assertions here. I'm thinking about pulling in https://github.com/sstephenson/bats to make tests better here.

**Pull request merged**